atlasdb-config/src/main/java/com/palantir/atlasdb/http/AtlasDbHttpClients.java

```diff
@@ -84,33 +85,39 @@ private AtlasDbHttpClients() {
      */
     public static <T> T createProxyWithFailover(
             Optional<SSLSocketFactory> sslSocketFactory,
+            Optional<ProxySelector> proxySelector,
             Collection<String> endpointUris,
             Class<T> type) {
         return createProxyWithFailover(
                 sslSocketFactory,
+                proxySelector,
                 endpointUris,
                 type,
                 UserAgents.DEFAULT_USER_AGENT);
     }
 
     public static <T> T createProxyWithFailover(
             Optional<SSLSocketFactory> sslSocketFactory,
+            Optional<ProxySelector> proxySelector,
             Collection<String> endpointUris,
             Class<T> type,
             String userAgent) {
```

Comment:
The original methods are part of the public API and are used e.g. in internal Atlas coordination product, so we should leave them as they are (maybe mark them as @Deprecated) and have them call the new methods (with proxyselector) with an empty Optional.

Reply:
should the method name stay the same for the new methods ? (they will be overloaded with different parameter set)

Reply:
Yep, overloading is fine - we do this pretty often in Atlas